### PR TITLE
Unbreak parser tests

### PR DESCRIPTION
**Pull request by moio**
Parser unit tests are at this point broken because of refactorings in the main codebase.

This PR fixes them and also two actual main codebase bugs discovered when tests were actually run.

### CODE REVIEW

File: controller/parser/multicast_request_parser.go [hunk 1]
```diff
@@ -87,8 +87,11 @@ func resolveArgsByServer(serverIDs []int64, allServerArgs []interface{}) (map[in
 
 func removeNamespace(method string) string {
 	parts := strings.Split(method, ".")
-	slice := parts[1:len(parts)]
-	return strings.Join(slice, ".")
+	if len(parts) > 1 {
```
Comment (chiaradiamarcelo) [inline controller/parser/multicast_request_parser.go]: shouldn't we fail with an error here if this has no "multicast" namespace?

Comment (moio) [reply controller/parser/multicast_request_parser.go]: See the commit message:

Namespaces are optional according to the XMLRPC standard

http://xmlrpc.com/spec.md

(and actually are Uyuni-specific, as dots per se have no meaning in
XMLRPC method names)

**Review by chiaradiamarcelo — APPROVED** [review]
Great, thanks! Just left a small comment.

File: controller/parser/multicast_request_parser.go (hunk 1 above)
Comment (chiaradiamarcelo) [reply controller/parser/multicast_request_parser.go]: True. But in this app, multicasting is executed if and only if the "multicast" namespace is present. So I guess if we got here and there's no "multicast" namespace something went wrong for sure.

**Review by chiaradiamarcelo — APPROVED** [review]

File: controller/parser/multicast_request_parser.go (hunk 1 above)
Comment (moio) [reply controller/parser/multicast_request_parser.go]: Are you suggesting the test is wrong? Currently it only calls `"method"`

**Review by chiaradiamarcelo — APPROVED** [review]

**Pull request merged**